nodebuilder/module.go

```diff
@@ -30,10 +30,10 @@ func ConstructModule(tp node.Type, cfg *Config, store Store) fx.Option {
 		// modules provided by the node
 		p2p.ConstructModule(tp, &cfg.P2P),
 		state.ConstructModule(tp, &cfg.State),
-		header.ConstructModule(tp, &cfg.Header),
 		share.ConstructModule(tp, &cfg.Share),
 		rpc.ConstructModule(tp, &cfg.RPC),
 		core.ConstructModule(tp, &cfg.Core),
+		header.ConstructModule(tp, &cfg.Header),
```

Comment:
REQ: for things like this, where ordering is important, we definitely need some sort of comment to make that explicit so that we don't change this in the future and spend hours/days debugging the same thing. 

Do we know if it is that `header` needs to be after `share`, `rpc`, and `core`? Or just `core`?

Reply:
> Do we know if it is that header needs to be after share, rpc, and core? Or just core?

It needs to be after `rpc` as per our investigations.